internet-indicator@sangorys/files/internet-indicator@sangorys/applet.js

```diff
@@ -0,0 +1,291 @@
+    _new_freq(){
+    	global.log(this.update_interval);
+        if (this._updateLoopID) {
+			Mainloop.source_remove(this._updateLoopID);
+		}
+        this._update_loop();
+    }
+
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    _get_status(){
+        let status = this._run_cmd("nordvpn status");
+        let result = status.split("\n")[0].split(": ")[1];
+        let outString;
+        if (result === "Connected"){
+            this.connected = true;
+            outString = "ON";
+        }else if (result === "Disconnected"){
+            this.connected = false;
+            outString = "OFF";
+        }else{
+            this.connected = false,
+            outString = "..."
+        }
+        this.set_applet_label(outString);
+    }
+
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    _update_loop() {
+		this._send_ping();	
+		//Util.spawnCommandLine("/usr/sbin/notify-send Applet '" + this.update_interval + "'"); //OK
+		//global.logError(this.update_interval);
+		this._updateLoopID = Mainloop.timeout_add(this.update_interval * 1000, Lang.bind(this, this._update_loop));
+	}
+
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    _send_ping() {
+        
+		// INIT
+		let [result, stdout, stderr] = ["", "", ""];
+		nbTry = nbTry + 1;
+		//Util.spawnCommandLine("/usr/sbin/notify-send Applet 'Scan " + nbTry + "'");
+		printDebug("Scan " + nbTry + " (period=" + this.update_interval + "s)");
+
+		
+        //////////////////////////////////////////////
+        // RUN PING
+		try {
+			[result, stdout, stderr] = GLib.spawn_command_line_sync("ping www.google.fr -c 1 -w 1000");
+			//printDebug("result=" + result);
+			//printDebug("stderr=" + stderr);
+			//printDebug("stdout=" + stdout);
+			//printDebug("oldStderr=" + oldStderr);
+		  }
+		catch (e) {
+			global.logError(e);
+		}
+		
+        
+        //////////////////////////////////////////////
+        // ANALYZE THE PING ANSWER
+		//global.logError("analyse");  
+		//global.logError(stdout.toString());  
+		//global.logError(stderr.toString().length);
+		
+		if (stderr == "")
+		{
+			//global.logError("ping OK");  
+			if (stderr.toString() != oldStderr)
+			{
+			  //global.logError("connecté");
+			  oldStderr = stderr.toString();
+			  Util.spawnCommandLine("/usr/sbin/notify-send 'Internet OK'");
+              this.set_applet_tooltip(_("Internet is connected :)"));
+			  this.update_interval = this.update_interval_when_internet;
+			  this.set_applet_icon_name("internet");
+			}
+			//else {global.logError("Pas de changement : connecté");}
+		}
+		else
+		{
+			//global.logError("ping failed");  
+			if (stderr.toString() != oldStderr)
+			{
+			  //global.logError("déconnecté");
+			  oldStderr = stderr.toString();
+			  Util.spawnCommandLine("/usr/sbin/notify-send 'Pas d internet'");
```

Comment:
You are sending an untranslated string to notify in French

Reply:
My bad. I fixed it